subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -73,22 +73,54 @@ struct k_thread prio_recv_thread_data;
 static K_KERNEL_STACK_DEFINE(prio_recv_thread_stack,
 			     CONFIG_BT_CTLR_RX_PRIO_STACK_SIZE);
 struct k_thread recv_thread_data;
-static K_KERNEL_STACK_DEFINE(recv_thread_stack, CONFIG_BT_RX_STACK_SIZE);
+static K_KERNEL_STACK_DEFINE(recv_thread_stack, CONFIG_BT_CTLR_RX_PRIO_STACK_SIZE);
```

Comment:
@cvinayak I had to reduce RAM use to build on BBC Microbit. This thread is no longer used for application callbacks, so I copied the required stack size from `prio_recv_thread`. Will this be enough stack size?

Reply:
I missed this comment here. 

This has serious consequence for the Zephyr Controller.... the stack size is reduced from 768 bytes to 448 bytes and is near overflow:
```
  BT RX               : STACK: unused 16 usage 432 / 448 (96 %); CPU: 0 %
      : Total CPU cycles used: %llu
 BT RX pri           : STACK: unused 280 usage 168 / 448 (37 %); CPU: 0 %
      : Total CPU cycles used: %llu
```

microbit have overlay files, the `CONFIG_BT_RX_STACK_SIZE`  can be adjusted there.

Reply:
Now that `CONFIG_BT_RECV_BLOCKING` is removed, this stack is not used much in the host. I see no point in keeping it as large as `CONFIG_BT_RECV_BLOCKING`? I think the stack watermark will probably be in controller code now, so the controller can define the stack size internally. Do we have a define for this, or could we define one?

Reply:
i still see `CONFIG_BT_RX_STACK_SIZE` is used as the `rx_thread_stack` size in core stack, this should be the value that controller too uses under `CONFIG_BT_HCI_RAW=y` (controller only builds). I do not see need to define a new Kconfig different from the core Host used one.

Reply:
You need a kconfig for the controller. Piggy-backing on the host kconfigs for setting stack sizes that depend on calls in the controller doesn't seem appropriate.